sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/async_challenge_auth_policy.py

```diff
@@ -69,6 +71,15 @@ async def on_challenge(self, request: "PipelineRequest", response: "PipelineResp
         except ValueError:
             return False
 
+        if self._verify_challenge_resource:
+            resource_domain = urlparse(scope).netloc
+            if not resource_domain:
+                raise ValueError(f"The challenge contains invalid scope '{scope}'.")
+
+            request_domain = urlparse(request.http_request.url).netloc
+            if not request_domain.lower().endswith(f".{resource_domain.lower()}"):
+                raise ValueError(f"The challenge resource '{resource_domain}' does not match the requested domain.")
```

Comment:
We want this to be a ValueError? As opposed to some kind of AuthenticationError?
I guess that's technically correct, though I'm wondering if there would be any benefit to users to be able to see the response that caused the failure like they could if this was some flavour of HttpResponseError.

Reply:
That's a good point, seeing the offending response would probably be helpful here. The response code at this point would be a 401, so raising a `ClientAuthenticationError` wouldn't be inconsistent. Do you think that would be an appropriate exception to raise in this case?

Reply:
Not sure about your pipeline, but I would hope authentication headers are sanitized so I'm not sure this would help. Besides, the response itself wouldn't be an error, so I not sure any sort of response-based exception makes sense.

Reply:
We don't sanitize anything from the response that's included in the HttpResponseError - but it's true that the response in that case is not actually an error status, so probably doesn't make much sense...

Reply:
Hm. I think I'll stick with `ValueError` for this release if there are no complaints, and we can revisit this when the issue is less pressing. I'm about to update the error message to include remediation steps (e.g. setting `verify_challenge_resource`), which more clearly indicate that a `ValueError` is appropriate, if not perfect